u8timeseries/backtesting/backtesting.py

```diff
@@ -259,3 +260,193 @@ def plot_residuals_analysis(residuals: TimeSeries, num_bins: int = 20):
+    by using the `backtest_forecast` function as a subroutine on `train_series`. This means that
+    the model is both trained and evaluated on `train_series`.
+    Note that the model is retrained for every single prediction, thus this mode is slower.
+
+    Split window mode (activated when `val_series` is passed):
+    This mode will be used when the `val_series` argument is passed.
+    For every hyperparameter combination, the model is trained on `train_series` and
+    evaluated on `val_series`.
+
+
+    Parameters
+    ----------
+    model
+        The ForecastingModel subclass to be tuned for 'series'.
+    parameters
+        A dictionary containing as keys hyperparameter names, and as values lists of values for the
+        respective hyperparameter.
+    train_series
+        The TimeSeries instance used for training (and also validation in split mode).
+    test_series
+        The TimeSeries instance used for validation in split mode.
+    fcast_horizon_n
+        The integer value of the forecasting horizon used in expanding window mode.
+    num_predictions:
+        The number of train/prediction cycles performed in one iteration of expanding window mode.
+    metric:
+        A function that takes two TimeSeries instances as inputs and returns a float error value.
+    verbose:
+        Whether to print progress.
+
+    Returns
+    -------
+    ForecastingModel
+        A 'model_class' instance with the best-performing hyperparameters from the given selection.
+    """
+
+    backtest_start_time = train_series.end_time() - (num_predictions + fcast_horizon_n) * train_series.freq()
+    min_error = float('inf')
+    best_param_combination = {}
+
+    # compute all hyperparameter combinations from selection
+    params_cross_product = list(product(*parameters.values()))
+
+    # iterate through all combinations of the provided parameters and choose the best one
+    iterator = _build_tqdm_iterator(params_cross_product, verbose)
+    for param_combination in iterator:
+        param_combination_dict = dict(list(zip(parameters.keys(), param_combination)))
+        model = model_class(**param_combination_dict)
+        if (val_series is None):  # expanding window mode
+            backtest_forecast = backtest_forecasting(train_series, model, backtest_start_time, fcast_horizon_n)
+            error = metric(backtest_forecast, train_series)
+        else:  # split mode
+            model.fit(train_series)
+            error = metric(model.predict(len(val_series)), val_series)
+        if (error < min_error):
+            min_error = error
+            best_param_combination = param_combination_dict
+    logger.info('Chosen parameters: ' + str(best_param_combination))
+    return model_class(**best_param_combination)
+
+
+def explore_models(train_series: TimeSeries,
+                   val_series: TimeSeries,
+                   test_series: TimeSeries,
+                   width: int = 3,
```

Comment:
I would move this parameter after `metric`, since it is more accessory and `metric` is really core to the logic. Also, I would maybe rename it `plot_width` or something like this, to make it clear it's only something for the plots.